perun-core/src/main/java/cz/metacentrum/perun/core/api/AuthzResolver.java

```diff
@@ -49,11 +50,26 @@ public static boolean isAuthorized(PerunSession sess, Role role, PerunBean compl
 	public static boolean isAuthorizedForAttribute(PerunSession sess, ActionType actionType, AttributeDefinition attrDef, Object primaryHolder, Object secondaryHolder) throws InternalErrorException {
 		try {
 			return AuthzResolverBlImpl.isAuthorizedForAttribute(sess, actionType, attrDef, primaryHolder, secondaryHolder);
-		} catch (AttributeNotExistsException | ActionTypeNotExistsException ex) {
+		} catch (AttributeNotExistsException | WrongAttributeAssignmentException ex) {
 			throw new InternalErrorException(ex);
 		}
 	}
 
+	/**
+	 * Return list of roles which are authorized for doing "action" on "attribute".
+	 *
+	 * @param sess perun session
+	 * @param actionType type of action on attribute (ex.: write, read, etc...)
+	 * @param attrDef attribute what principal want to work with
+	 * @return list of roles
+	 * @throws InternalErrorException
+	 * @throws AttributeNotExistsException
+	 * @throws ActionTypeNotExistsException
+	 */
+	public static List<Role> getRolesWhichCanWorkWithAttribute(PerunSession sess, ActionType actionType, AttributeDefinition attrDef) throws InternalErrorException, AttributeNotExistsException, ActionTypeNotExistsException {
```

Comment:
I don't see any usage of this. Is it a planned feature ? Because we have also #1970, which modify this method in Bl and would probably need a rebase after merging.

Reply:
You are right. Removed.